Created Basic Autonomous

eandr127 opened the pull request:
This still needs to be tested, and should also be slimmed down a lot

eandr127 commented:
I should also remove ERobotDrive

jaimeyu commented on src/ca/team2706/frc/autonomous/AutoHelper.java:
```diff
@@ -0,0 +1,48 @@
+	public static boolean writeLineToFile(String line, File file) {
+		PrintStream stream = null;
+		try {
+			stream = new PrintStream(new FileOutputStream(file, false));
```
Is there a close for the file descriptor or are we allowing the garbage collector to handle this?
What is the behaviour when multiple back to back calls to writeLineToFile happen and the file descriptor is held open?

eandr127 replied:
Yes, it should have been closed, but I don't think we really ran into any problems with it since we only calibrated the robot a couple times.

jaimeyu commented on src/ca/team2706/frc/autonomous/AutoHelper.java:
```diff
@@ -0,0 +1,48 @@
+		PrintStream stream = null;
+		try {
+			stream = new PrintStream(new FileOutputStream(file, false));
+		} catch (FileNotFoundException e) {
```
Interesting, who firsts creates the file? Could we create the file if we can't find it?

eandr127 replied:
I think it needed to be added manually, but we should create it if we can't find it.

jaimeyu commented on src/ca/team2706/frc/autonomous/AutoModeSelector.java:
```diff
@@ -0,0 +1,96 @@
+				newInputs[i] = !inputs[i].get();
+				System.out.println("Input " + i + " = " + newInputs[i]);
+			}
+			// TODO: encodeBools() works. Verified by test. So that means there's 
```
What's funny?

eandr127 replied:
We used dipswitches to choose an autonomous, and we did have a couple problems with wiring them, and figuring out which side to start from, but I'm not entirely sure.

eandr127 replied:
Do you know if we will have a dipswitch this year?

jaimeyu commented on src/ca/team2706/frc/autonomous/commands/DriveStraightAndTurn.java:
```diff
@@ -0,0 +1,107 @@
+	@Override
+	public boolean tick() {
+		if (!Subsystems.gyroPID.isDone()) {
+			System.out.println("gyro.getAngle() = " + Subsystems.gyroSensor.getAngle());
```
We should wrap the system.outs with a debug conditional. The system.outs are time expensive and eat up our bandwidth as they have to go over wifi.

eandr127 replied:
Yes, that's a very good idea.

jaimeyu replied:
We can optionally not drop the message and pipe it to a file instead of /dev/null. I don't think we're hurting for performance but the vision team might want the extra bandwidth over the wifi. And we still get to keep the logs so even during a match, we're still capturing data.

eandr127 replied:
So we always print to a file on the robot, and only if in debug mode, print to the Riolog?

jaimeyu replied:
that sounds like a plan. maybe look into log4j so it can handle log
rotation so we dont run out of disk space.

--
Sincerely,
Jaime Yu
B.Eng in Computer Engineering, System Hardware
Software Developer III at Juniper Networks

$CV = "http://cv.jaimeyu.com"
$Blog = "http://ask.jaimeyu.com"
$Project = "http://www.capstone490.com"
$Linkedin = "https://www.linkedin.com/in/jaimeyu"
$GitHub = "https://github.com/jaimeyu"

On Wed, Nov 11, 2015 at 7:04 PM, eandr127 <notifications@github.com> wrote:

> In src/ca/team2706/frc/autonomous/commands/DriveStraightAndTurn.java
> <https://github.com/FRC2706/2706-Autonomous/pull/3#discussion_r44605804>:
>
> > +		// Reset the gyro PID to a reasonable state.
> > +		Subsystems.gyroPID.resetErrorSum();
> > +		Subsystems.gyroPID.resetPreviousVal();
> > +		// Used to make sure that the PID doesn't bail out as done
> > +		// right away (we know the gyro angle is zero from the above
> > +		// reset).
> > +		Subsystems.gyroPID.calcPID(0);
> > +		
> > +
> > +		
> > +	}
> > +
> > +	@Override
> > +	public boolean tick() {
> > +		if (!Subsystems.gyroPID.isDone()) {
> > +			System.out.println("gyro.getAngle() = " + Subsystems.gyroSensor.getAngle());
>
> So we always print to a file on the robot, and only if in debug mode,
> print to the Riolog?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FRC2706/2706-Autonomous/pull/3/files#r44605804>.
>

eandr127 replied:
Ok, sure

eandr127 replied:
Do you think java.util.logging can do that too?

jaimeyu replied:
This look like what you want http://kodejava.org/how-do-i-create-a-rolling-log-files/

Log4j is really nice library. Used it in other projects to simplify our life.

eandr127 commented:
I've added some basic logging and removed some old code from last year.  I still need to implement a rolling file appender and make stdout only used when debugging.
